src/sync/sync_session.cpp

```diff
@@ -427,13 +427,19 @@ void SyncSession::handle_progress_update(uint64_t downloaded, uint64_t downloada
         m_current_downloadable = downloadable;
         m_current_downloaded = downloaded;
         m_current_uploaded = uploaded;
+        std::vector<uint64_t> to_erase;
         for (auto& pair : m_notifiers) {
             bool should_delete = false;
-            invocations.emplace_back(create_notifier_invocation(pair.second, should_delete));
+            auto package = pair.second;
+            invocations.emplace_back(create_notifier_invocation(package, should_delete));
             if (should_delete) {
-                m_notifiers.erase(pair.first);
+                to_erase.emplace_back(pair.first);
             }
         }
+        // We need to erase after iteration since our Linux builds break if we try it the C++14 way.
```

Comment:
I think it'd be better to state which compiler it breaks with, and why. Something like "std::unordered_map::erase in GCC 4.9's libstdc++ doesn't follow C++14's requirement that it preserve the order of the remaining elements, so we cannot safely erase the elements while iterating the map". (I'm speculating at the compiler version in question)